Tools/CISupport/ews-build/steps.py

```diff
@@ -959,54 +1051,105 @@ def skip_build(self, reason):
 
     def start(self):
         patch_id = self.getProperty('patch_id', '')
-        if not patch_id:
-            self._addToLog('stdio', 'No patch_id found. Unable to proceed without patch_id.\n')
-            self.descriptionDone = 'No patch id found'
+        pr_number = self.getProperty('github.number', '')
+
+        if not patch_id and not pr_number:
+            self._addToLog('stdio', 'No patch_id or pr_number found. Unable to proceed without one of them.\n')
+            self.descriptionDone = 'No change found'
+            self.finished(FAILURE)
+            return None
+
+        if patch_id and pr_number:
+            self._addToLog('stdio', 'Both patch_id and pr_number found. Unable to proceed with both.\n')
+            self.descriptionDone = 'Two changes found'
```

Comment:
This will show up in UI, maybe better to show something like: "Error: both PR and patch number found"